cloud/vmware/vmware_vm_shell.py

```diff
@@ -0,0 +1,191 @@
+options:
+    hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter API server
+        required: True
+    username:
+        description:
+            - The username of the vSphere vCenter
+        required: True
+        aliases: ['user', 'admin']
+    password:
+        description:
+            - The password of the vSphere vCenter
+        required: True
+        aliases: ['pass', 'pwd']
+    datacenter:
+        description:
+            - The datacenter hosting the VM
+        required: False
+    vm_id:
+        description:
+            - The identification for the VM
+        required: True
+    vm_id_type:
+        description:
+            - The identification tag for the VM
+        default: dns_name
+        choices:
+            - 'uuid'
+            - 'dns_name'
+            - 'inventory_path'
+        required: False
+    vm_username:
+        description:
+            - The user to connect to the VM.
+        required: False
+    vm_password:
+        description:
+            - The password used to login to the VM.
+        required: False
+    vm_shell:
+        description:
+            - The absolute path to the program to start. On Linux this is executed via bash.
+        required: True
+    vm_shell_args:
+        description:
+            - The argument to the program. 
+        required: False
+        default: None
+    vm_shell_env:
+        description:
+            - Comma seperated list of envirnoment variable, specified in the guest OS notation
+        required: False
+        default: None
+    vm_shell_cwd:
+        description:
+            - The current working directory of the application from which it will be run
+        required: False
+        default: None
+'''
+
+EXAMPLES = '''
+    - name: shell execution
+      local_action:
+        module: vmware_vm_shell
+        hostname: myVSphere
+        username: myUsername
+        password: mySecret
+        datacenter: myDatacenter
+        vm_id: DNSnameOfVM
+        vm_username: root
+        vm_password: superSecret
+        vm_shell: /bin/echo
+        vm_shell_args: " $var >> myFile "
+        vm_shell_env:
+          - "PATH=/bin"
+          - "var=test"
+        vm_shell_cwd: "/tmp"
+
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+def find_vm(content, vm_id, vm_id_type="dns_name", datacenter=None):
+    si = content.searchIndex
+    vm = None
+
+    if datacenter:
+        datacenter = find_datacenter_by_name(content, datacenter)
+    
+    if vm_id_type == 'dns_name':
+        vm = si.FindByDnsName(datacenter=datacenter, dnsName=vm_id, vmSearch=True)
+    elif vm_id_type == 'inventory_path':
+        vm = si.FindByInventoryPath(inventoryPath=vm_id)
+        if type(vm) != type(vim.VirtualMachine):
+            vm = None
+    elif vm_id_type == 'uuid':
+        vm = si.FindByUuid(datacenter=datacenter, uuid=vm_id, vmSearch=True)
+
```

Comment:
Hi @ritzk  - I went ahead and merged this code, and added support for searching by vm_name - I am keeping it as the default just because that is how the other modules operate.  Good point, however, about the speed for searches using inventory paths...